packages/clay-multi-select/src/index.tsx

```diff
@@ -42,6 +58,12 @@ export interface IProps extends React.HTMLAttributes<HTMLInputElement> {
 	 */
 	closeButtonAriaLabel?: string;
 
+	/**
+	 * Adds a component to replace the default component that renders
+	 * the content of the `<ClayDropDown />` component.
+	 */
+	menuContentComponent?: MenuContentComponent;
```

Comment:
Not sure about that name, maybe `menuComponent` just?

Reply:
Should we be allowing customization of the whole menu, or just the menu items? Typically I think we would do a `*Renderer` type prop, which is just a function that takes the typical props expected for an item.

I think I would expect this to be only for items as `autocompleteItemRenderer`, the whole menu seems strange to me.

Reply:
Yeah I thought about that if it was just the Item, but the filter rule is stuck in our implementation so I let all the Menu content change so that they can have filter control.

I thought about leaving `*Component` to follow the props pattern elsewhere in our components.

Reply:
What if we did `autocompleteMenuRenderer` or just `menuRenderer` and it be a function? I lean towards using renderer functions rather than components since they are slightly more flexible but also allow for a component as well.

Reply:
Hmm, I'm not sure if I understand. But do you mean that going with `*Component` in the name of the prop can have a tone that can add Class components?

Reply:
I would go with `*Renderer` and have its type be something like...
```tsx
(props: any) => JSX.Element | React.ComponentType<any>
```

That make sense? This way users can pass either a component(function or class), or even just a function that returns jsx.

Reply:
Yeah makes sense. I changed it to `menuRenderer` and kept the type `React.FunctionComponent<IMenuRendererProps>;` I think it covers these cases.